src/catalog/database_catalog.cpp

```diff
@@ -1330,6 +1330,38 @@ std::vector<std::pair<common::ManagedPointer<storage::index::Index>, const Index
   return index_objects;
 }
 
+bool DatabaseCatalog::CopyPR(common::ManagedPointer<transaction::TransactionContext> txn,
+                             storage::ProjectedRow *table_pr, storage::ProjectedRow *index_pr, index_oid_t index_oid,
+                             common::ManagedPointer<terrier::storage::SqlTable> table,
+                             storage::TupleSlot table_tuple_slot) {
+  const auto key_schema = GetIndexSchema(txn, index_oid);
+  const auto index = GetIndex(txn, index_oid);
+  const auto &indexed_attributes = key_schema.GetIndexedColOids();
+  auto pr_map = table->ProjectionMapForOids(indexed_attributes);
+
+  auto result = table->Select(txn, table_tuple_slot, table_pr);
+  if (!result) {
+    return false;
```

Comment:
We should never fail to `Select` the results of an index scan since that performs visibility checks. We should have an assert on the result here.